src/Parser.hs

```diff
@@ -39,8 +39,7 @@ parserRoot =
   do
     noteChar <- oneOf "ABCDEFG"
     let note = read [noteChar]
-    acc <- parserAccidental
-    return $ Root note acc
+    Root note <$> parserAccidental
```

Comment:
This is a _really_ subtle refactor; its correctness depends on the algebraic laws of functors / monads. In effect, it says that instead of first parsing an accidental and then building a `Root` that's then lifted into the `Parser` monad, we can instead map the partially-applied function `Root note` over the result of running the `Accidental` parser.

Reply:
Reading this again, I'm dissatisfied with it. It doesn't make it transparent what that computation actually _means_; unfortunately, I'm not sure I can describe it without some more of the language of monads. I wrote up an introduction to monads for COMP105 last fall; I'll get that to you when I can (for some reason I can't attach it to this commented as intended.)